src/rust/catloop/mod.rs

```diff
@@ -568,9 +568,13 @@ impl CatloopLibOS {
         self.scheduler.poll()
     }
 
-    /// Takes out the [OperationResult] associated with the target [SchedulerHandle].
+    /// Takes out the [OperationResult] associated with the target [TaskHandle].
     fn take_result(&mut self, handle: TaskHandle) -> (QDesc, OperationResult) {
-        let task: OperationTask = OperationTask::from(self.scheduler.remove(handle).as_any());
+        let task: OperationTask = if let Some(task) = self.scheduler.remove(&handle) {
+            OperationTask::from(task.as_any())
+        } else {
+            panic!("Removing task that does not exist (either was previously removed or never inserted)");
```

Comment:
Shouldn't this condition be `unreachable!()`? Even better, wouldn't be the case to change the function signature to return a `Fail` and we propagate it back to the PDPIX call?